test/models/watching_transaction_test.rb

```diff
@@ -0,0 +1,139 @@
+    end
+
+    stub_post = stub_request(:post, EthereumApi::SERVER_URL)
+                  .with(body: /eth_getTransactionByHash/)
+                  .to_return(
+                    body: {
+                      result: {
+                        'block_number' => nil,
+                      }
+                    }.to_json
+                  )
+
+    WatchingTransaction.resend_transactions
+
+    assert_requested(stub_post, times: 5)
+    assert_equal 5, WatchingTransaction.where(group_id: original.group_id).count,
+                 'should work'
+  end
+
+  test 'resend transactions should destroy all if one is mined' do
+    original = FactoryBot.create(:watching_transaction)
+    3.times do
+      FactoryBot.create(:watching_transaction, group_id: original.group_id, user: original.user)
+    end
+    latest = FactoryBot.create(:watching_transaction, group_id: original.group_id, user: original.user)
+
+    stub_post = stub_request(:post, EthereumApi::SERVER_URL)
+                  .with(body: /eth_getTransactionByHash/)
+                  .to_return(
+                    body: {
+                      result: {
+                        'block_number' => nil,
+                      }
+                    }.to_json
+                  )
+
+    stub_post_one = stub_request(:post, EthereumApi::SERVER_URL)
+                      .with(body: /eth_getTransactionByHash.*#{latest.txhash}/)
+                      .to_return(
+                        body: {
+                          result: {
+                            'block_number' => SecureRandom.hex(8),
+                          }
+                        }.to_json
+                      )
+
+    WatchingTransaction.resend_transactions
+
+    assert_requested(stub_post, times: 5)
+    assert_requested(stub_post_one, times: 1)
+    assert_equal 0, WatchingTransaction.where(group_id: original.group_id).count,
+                 'should destroy all if one is mined'
+  end
+
+  test 'resend transactions should resend latest if rest are dropped' do
+    original = FactoryBot.create(:watching_transaction)
+    2.times do
+      FactoryBot.create(:watching_transaction, group_id: original.group_id, user: original.user)
+    end
+    sleep 1 # UUID is not sortable, so some delay is needed in order to properly sort via created_at
+    latest = FactoryBot.create(:watching_transaction, group_id: original.group_id, user: original.user)
+
+    stub_post = stub_request(:post, EthereumApi::SERVER_URL)
+      .with(body: /eth_getTransactionByHash/)
+      .to_return(
+        body: {
+          result: nil
+        }.to_json
+      )
+
+    new_txhash = generate(:txhash)
+    stub_post_send = stub_request(:post, EthereumApi::SERVER_URL)
+      .with(body: /eth_sendRawTransaction/)
+      .to_return(
+        body: {
+          result: new_txhash
+        }.to_json
+      )
+
+    WatchingTransaction.resend_transactions
+
+    assert_requested(stub_post, times: 4)
```

Comment:
Ditto with using depending on a variable size instead of  a fixed number the comments above